### PR TITLE
Spellcheck integration testing

### PR DESCRIPTION
**Pull request by cea2aj**
Convert a spellcheck acceptance test into integration tests

This test coverage has two parts. In the first part, our test coverage confirms that the spellcheck components renders with the correct query params in the hyperlink. To accomplish this, I converted the previous spellcheck tests to more of an integration test by mounting the component with the enzyme adapter.

In the second part, I confirm that when ANSWERS initializes with the spellcheck params in the URL, a search is fired with those spellcheck params. To make this possible, this PR sets up integration testing for answers-umd which previously did not exist in our jest tests. This PR mocks the core library so that we don't actually fire HTTP requests when running these tests.

J=SLAP-1062
TEST=auto

Add integration test coverage

### CODE REVIEW

**Review by tmeyer2115 — APPROVED** [review]
LGTM! Really like the mocked Core pattern. Maybe we can leverage that and the new `answers-umd` test suite to convert other acceptance tests over.

File: tests/answers-umd.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import ANSWERS from '../src/answers-umd';
+
+let windowSpy;
+
+describe('ANSWERS instance integration testing', () => {
+  beforeEach(() => {
+    windowSpy = jest.spyOn(window, 'window', 'get');
```
Comment (oshi97) [inline tests/answers-umd.js:7]: is this needed so we can reset the window after each test?

Comment (cea2aj) [reply tests/answers-umd.js:7]: ya, it's setup this way so that on each test we can have a fresh window to work with

**Review by oshi97 — APPROVED** [review]

**Comment by oshi97** [comment]
like the mock for the answers-core!

**Comment by cea2aj** [comment]
> like the mock for the answers-core!

I was surprised to find that without the mock, answers-core will actually fire real HTTP requests when this test is ran

**Pull request merged**